pages/docs/reference/fun-interfaces.md

```diff
@@ -0,0 +1,101 @@
+---
+type: doc
+layout: reference
+title: "Functional interfaces (SAM interfaces)"
+---
+
+# Functional (SAM) interfaces
+
+An interface with only one abstract method is called a _functional interface_, or a _Single Abstract 
```

Comment:
For other parts of doc (data classes, sealed classes...), the description starts from use case explanation.
I think we have a good example here https://kotlinlang.org/docs/reference/java-interop.html#sam-conversions

Something like:
```
Since 1.4, Kotlin supports functional interface conversions (SAM conversions). This means that Kotlin function 
literals can be automatically converted into implementations of _functional_ interfaces with a single abstract 
method, as long as the parameter types of the interface method match the parameter types of the Kotlin 
function.

To declare a functional interface, one puts the `fun` modifier before the interface:

<example>
```

Reply:
The more good realistic examples, the better. 
It's an example of a conversion. I would start with a functional interface with a good example and then explain conversions. Mixing one with another seems overloading and makes it harder to understand.

BTW, in the Internet, there are many materials about Kotlin that explain everything in easier ways with simple words. The easier we write, the more people will adopt it. 
Some language articles on our web site are too overloaded with words and complex grammatical constructions. I'm for simple language and explanation. :) 

As for explanation of a functional interface, what can we add (without conversions)?

Reply:
I have updated examples and wording.